src/webgpu/shader/execution/builtin/select.spec.ts

```diff
@@ -133,5 +141,97 @@ Please read the following guidelines before contributing:
 https://github.com/gpuweb/cts/blob/main/docs/plan_autogen.md
 `
   )
-  .params(u => u.combine('placeHolder1', ['placeHolder2', 'placeHolder3']))
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('component', ['b', 'f', 'i', 'u'] as const)
+      .combine('overload', ['vec2', 'vec3', 'vec4'] as const)
+  )
+  .fn(async t => {
+    const componentType = dataType[t.params.component as scalarKind].type;
+    const cons = dataType[t.params.component as scalarKind].constructor;
+
+    // Create the scalar values that will be selected from, either as scalars
+    // or vectors.
+    //
+    // Each boolean will select between c[k] and c[k+4].  Those values must
+    // always compare as different.  The tricky case is boolean, where the parity
+    // has to be different, i.e. c[k]-c[k+4] must be odd.
+    const c = [0, 1, 2, 3, 5, 6, 7, 8].map(i => cons(i)) as Scalar[];
+    const T = True;
+    const F = False;
+    // Form vectors be used for the 'false' and 'true' data operands.
+    // Each component must be different in the FF* vectors must be different
+    // from the corresponding component in the TT* vector of the same length.
+    const FF = vec2(c[0], c[1]);
```

Comment:
This works, but I got confused over `T` and `F` being aliases for `True` and `False`, where as the `TT` and `FF` aliases are not to booleans, but the first two operands.
The `pickN` functions also provide limited simplification. Given that you've fully expanded the case list (which I love), it doesn't seem any more verbose to build the expected vectors without any helpers.

If you add the following to the `Vector` class:

```ts
  public get x() {
    return this.elements[0];
  }
  public get y() {
    return this.elements[1];
  }
  public get z() {
    return this.elements[2];
  }
  public get w() {
    return this.elements[3];
  }
```

Then you could be entirely explicit, without any helpers:

```ts
    // Create the scalar values that will be selected from, either as scalars
    // or vectors.
    //
    // Each boolean will select between c[k] and c[k+4].  Those values must
    // always compare as different.  The tricky case is boolean, where the parity
    // has to be different, i.e. c[k]-c[k+4] must be odd.
    const c = [0, 1, 2, 3, 5, 6, 7, 8].map(i => cons(i)) as Scalar[];
    const T = True;
    const F = False;

    var tests: { dataType: VectorType; boolType: VectorType; cases: CaseList };

    switch (t.params.overload) {
      case 'vec2': {
        const a = vec2(c[0], c[1]);
        const b = vec2(c[4], c[5]);
        tests = {
          dataType: TypeVec(2, componentType),
          boolType: TypeVec(2, TypeBool),
          cases: [
            { input: [a, b, vec2(F, F)], expected: vec2(a.x, a.y) },
            { input: [a, b, vec2(F, T)], expected: vec2(a.x, b.y) },
            { input: [a, b, vec2(T, F)], expected: vec2(b.x, a.y) },
            { input: [a, b, vec2(T, T)], expected: vec2(b.x, b.y) },
          ],
        };
        break;
      }
      case 'vec3': {
        const a = vec3(c[0], c[1], c[2]);
        const b = vec3(c[4], c[5], c[6]);
        tests = {
          dataType: TypeVec(3, componentType),
          boolType: TypeVec(3, TypeBool),
          cases: [
            { input: [a, b, vec3(F, F, F)], expected: vec3(a.x, a.y, a.z) },
            { input: [a, b, vec3(F, F, T)], expected: vec3(a.x, a.y, b.z) },
            { input: [a, b, vec3(F, T, F)], expected: vec3(a.x, b.y, a.z) },
            { input: [a, b, vec3(F, T, T)], expected: vec3(a.x, b.y, b.z) },
            { input: [a, b, vec3(T, F, F)], expected: vec3(b.x, a.y, a.z) },
            { input: [a, b, vec3(T, F, T)], expected: vec3(b.x, a.y, b.z) },
            { input: [a, b, vec3(T, T, F)], expected: vec3(b.x, b.y, a.z) },
            { input: [a, b, vec3(T, T, T)], expected: vec3(b.x, b.y, b.z) },
          ],
        };
        break;
      }
      case 'vec4': {
        const a = vec4(c[0], c[1], c[2], c[3]);
        const b = vec4(c[4], c[5], c[6], c[7]);
        tests = {
          dataType: TypeVec(4, componentType),
          boolType: TypeVec(4, TypeBool),
          cases: [
            { input: [a, b, vec4(F, F, F, F)], expected: vec4(a.x, a.y, a.z, a.w) },
            { input: [a, b, vec4(F, F, F, T)], expected: vec4(a.x, a.y, a.z, b.w) },
            { input: [a, b, vec4(F, F, T, F)], expected: vec4(a.x, a.y, b.z, a.w) },
            { input: [a, b, vec4(F, F, T, T)], expected: vec4(a.x, a.y, b.z, b.w) },
            { input: [a, b, vec4(F, T, F, F)], expected: vec4(a.x, b.y, a.z, a.w) },
            { input: [a, b, vec4(F, T, F, T)], expected: vec4(a.x, b.y, a.z, b.w) },
            { input: [a, b, vec4(F, T, T, F)], expected: vec4(a.x, b.y, b.z, a.w) },
            { input: [a, b, vec4(F, T, T, T)], expected: vec4(a.x, b.y, b.z, b.w) },
            { input: [a, b, vec4(T, F, F, F)], expected: vec4(b.x, a.y, a.z, a.w) },
            { input: [a, b, vec4(T, F, F, T)], expected: vec4(b.x, a.y, a.z, b.w) },
            { input: [a, b, vec4(T, F, T, F)], expected: vec4(b.x, a.y, b.z, a.w) },
            { input: [a, b, vec4(T, F, T, T)], expected: vec4(b.x, a.y, b.z, b.w) },
            { input: [a, b, vec4(T, T, F, F)], expected: vec4(b.x, b.y, a.z, a.w) },
            { input: [a, b, vec4(T, T, F, T)], expected: vec4(b.x, b.y, a.z, b.w) },
            { input: [a, b, vec4(T, T, T, F)], expected: vec4(b.x, b.y, b.z, a.w) },
            { input: [a, b, vec4(T, T, T, T)], expected: vec4(b.x, b.y, b.z, b.w) },
          ],
        };
        break;
      }
    }

    run(
      t,
      'select',
      [tests.dataType, tests.dataType, tests.boolType],
      tests.dataType,
      t.params,
      tests.cases
    );
```

WDYT?

Reply:
Yeah, sorry about the FF etc being different underlying types.

I had not yet learned / internalized the getter feature.  
Your version is better.